packages/jest-matchers/src/index.js

```diff
@@ -276,15 +265,14 @@ expect.extend(toThrowMatchers);
 
 expect.addSnapshotSerializer = () => void 0;
 expect.assertions = (expected: number) => {
-  global[GLOBAL_STATE].state.expectedAssertionsNumber = expected;
+  getState().expectedAssertionsNumber = expected;
 };
 expect.hasAssertions = expected => {
   utils.ensureNoExpected(expected, '.hasAssertions');
-  global[GLOBAL_STATE].state.isExpectingAssertions = true;
-};
-expect.setState = (state: Object) => {
-  Object.assign(global[GLOBAL_STATE].state, state);
+  getState().isExpectingAssertions = true;
 };
-expect.getState = () => global[GLOBAL_STATE].state;
+expect.getState = getState;
+expect.setState = setState;
+expect.extractExpectedAssertionsErrors = extractExpectedAssertionsErrors;
```

Comment:
@thymikee yeah! i agree for `assert` support, but `expect.assertions` is another story, since it's an `expect` feature and heavily coupled to matchers package